clients/lsp-eslint.el

```diff
@@ -253,10 +253,30 @@ source.fixAll code action."
         first-argument-exist
       (executable-find (cl-first eslint-server-command)))))
 
-(defun lsp-eslint--confirm-local (_workspace _params)
-  ;; don't bother implementing since it will be dropped in next version of the
-  ;; server
-  t)
+(defvar lsp-eslint--stored-libraries (ht)
```

Comment:
nit: make this public if we are going to use it in combination with desktop-mode

Reply:
I've added it to `desktop-globals-to-save` and added a `(require 'desktop)` at the top for persistence - is this the right way to do that?

Reply:
If we are going to use desktop-save-mode it should be in the docs, not in the code. If we do it in the code we will force heavy dependency unconditionally.

You may implement the persistence similar to what we do for the session, check `lsp--persist-session`.

Reply:
Ahh, gotcha. Was talking a look at `lsp--persist-session` and noticed that sessions have a `metadata` field on them - would that be suitable for storing this information?

Reply:
In general, yes, but the issue is that dap stores in the metadata information whcih cannot be persisted. As you can see, lsp--persist-session strips the metadata. We may strip the dap-specific stuff though and use that field.

Reply:
What you can do is to use the metadata in this PR and later we can fix dap-mode issue.

Reply:
Bah, just missed that comment - I implemented it using `lsp--persist` and `lsp--read-from-file`. Which one do we want to go for?

Reply:
I am personally fine with this solution(in general, the requirements are less strict for non-core files). The only issue is that we are creating more work for https://github.com/emacscollective/no-littering . I can merge it as it is.

Reply:
I can merge it as it is.

Reply:
Sounds good to me. We can move it to metadata once the dap-mode issue is fixed.